sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -307,6 +303,151 @@ export class ApplicationService extends RecordDataModelService<Application> {
+   * @param applicationData application dynamic data to be saved.
+   * @param associatedFiles files to be associated with the application.
+   * @param auditUserId user that should be considered the one that is making the change request.
+   * @returns application ID of the created application that represents the change request.
+   */
+  async submitApplicationChangeRequest(
+    applicationId: number,
+    studentId: number,
+    applicationData: ApplicationData,
+    associatedFiles: string[],
+    auditUserId: number,
+  ): Promise<ApplicationSubmissionResult> {
+    const application = await this.getApplicationToSave(
+      studentId,
+      ApplicationStatus.Completed,
+      applicationId,
+    );
+    // If the application was not found it is because it does not belong to the student or it is not completed.
+    if (!application) {
+      throw new CustomNamedError(
+        "Student application not found or it is not in the correct status to be changed.",
+        APPLICATION_NOT_FOUND,
+      );
+    }
+    if (!allowApplicationChangeRequest(application.programYear)) {
+      throw new CustomNamedError(
+        "The program year associated to this application does not allow a change request submission.",
+        APPLICATION_NOT_VALID,
+      );
+    }
+    // Check if there is already a change request in progress.
+    const hasChangeInProgress = application.parentApplication.versions.some(
+      (version) =>
+        [
+          ApplicationEditStatus.ChangeInProgress,
+          ApplicationEditStatus.ChangePendingApproval,
+        ].includes(version.applicationEditStatus),
+    );
+    if (hasChangeInProgress) {
+      throw new CustomNamedError(
+        "An application change request is already in progress.",
+        APPLICATION_CHANGE_REQUEST_ALREADY_IN_PROGRESS,
+      );
+    }
+    // Validates selected location and selected offering as key values that
+    // should not be changed. Even if some other values are changed (tampered by the user),
+    // there will be no impact for application processing since the location and offering
+    // are copied from the current application retrieved from the database.
+    if (
+      applicationData.selectedLocation !== application.data.selectedLocation
+    ) {
+      throw new CustomNamedError(
+        "Change request has a different location from its original submission.",
+        APPLICATION_NOT_VALID,
+      );
+    }
+    if (
+      applicationData.selectedOffering !== application.data.selectedOffering
+    ) {
+      // Applications are expected to have the same offering value populated, even for PIRs.
+      // PIRs will not have the offering value populated and the assessment will be created
+      // based on the most recent offering associated with the current application version.
+      throw new CustomNamedError(
+        "Change request has a different offering from its original submission.",
+        APPLICATION_NOT_VALID,
+      );
+    }
+
```

Comment:
I believe we are missing the check for application to be archived unless there is a reason that I am missing. Current change request process validates archived applications.

Reply:
There are no requirements for it.

Reply:
But is not considered as an existing direction based on the current change request validation?

Reply:
> But is not considered as an existing direction based on the current change request validation?

Short answer, no.

Reply:
I sent a message Josh, but unless it is a clear requirement, it should not be part of this PR and assuming that it should be part because the current implementation of the appeals may be a mistake.